archunit/src/main/java/com/tngtech/archunit/core/domain/properties/HasParameters.java

```diff
@@ -0,0 +1,109 @@
+ * limitations under the License.
+ */
+package com.tngtech.archunit.core.domain.properties;
+
+import java.util.List;
+
+import com.tngtech.archunit.PublicAPI;
+import com.tngtech.archunit.base.DescribedPredicate;
+import com.tngtech.archunit.core.domain.JavaCodeUnit;
+import com.tngtech.archunit.core.domain.JavaParameter;
+
+import static com.tngtech.archunit.PublicAPI.Usage.ACCESS;
+
+public interface HasParameters extends HasParameterTypes {
+    /**
+     * @return the {@link JavaParameter parameters} of this {@link JavaCodeUnit}. On the contrary to the Reflection API this will only contain
+     *         the parameters from the signature and not synthetic parameters, if the signature is generic. In these cases
+     *         {@link #getParameters()}{@code .size()} will always be equal to {@link #getParameterTypes()}{@code .size()},
+     *         but not necessarily to {@link #getRawParameterTypes()}{@code .size()} in case the compiler adds synthetic parameters.<br>
+     *         Note that for non-generic method signatures {@link #getParameters()} actually contains the raw parameter types and thus
+     *         can also contain synthetic parameters. Unfortunately there is no way at the moment to distinguish synthetic
+     *         parameters from non-synthetic parameters in these cases.
+     *
+     * @see #getRawParameterTypes()
+     * @see #getParameterTypes()
+     */
+    @PublicAPI(usage = ACCESS)
+    List<JavaParameter> getParameters();
+
+    /**
+     * Predefined {@link DescribedPredicate predicates} targeting objects that implement {@link HasParameters}
+     */
+    @PublicAPI(usage = ACCESS)
+    final class Predicates {
+        private Predicates() {
+        }
+
+        /**
+         * @param predicate A {@link DescribedPredicate} for the {@link JavaParameter}s of the {@link HasParameters}
+         * @return A {@link DescribedPredicate predicate} for a {@link HasParameters} that returns {@code true}
+         *         if and only if at least one {@link JavaParameter} of the {@link HasParameters} matches the given predicate.
+         */
+        @PublicAPI(usage = ACCESS)
+        public static DescribedPredicate<HasParameters> anyParameterThat(DescribedPredicate<? super JavaParameter> predicate) {
+            return new AnyParameterPredicate(predicate);
+        }
+
+        /**
+         * @param predicate A {@link DescribedPredicate} for the {@link JavaParameter}s of the {@link HasParameters}
+         * @return A {@link DescribedPredicate predicate} for a {@link HasParameters} that returns {@code true}
+         *         if and only if all {@link JavaParameter}s of the {@link HasParameters} match the given predicate.
+         */
+        @PublicAPI(usage = ACCESS)
+        public static DescribedPredicate<HasParameters> allParameters(DescribedPredicate<? super JavaParameter> predicate) {
+            return new AllParametersPredicate(predicate);
```

Comment:
You could reuse existing functionality:
```suggestion
            return DescribedPredicate.<JavaParameter>allElements(predicate).onResultOf(HasParameters::getParameters)
                    .as("all parameters " + predicate.getDescription());
```